api/v1alpha1/extendeddaemonset_types.go

```diff
@@ -75,6 +75,7 @@ type ExtendedDaemonSetSpecStrategyRollingUpdate struct {
 type ExtendedDaemonSetSpecStrategyCanary struct {
 	Replicas     *intstr.IntOrString   `json:"replicas,omitempty"`
 	Duration     *metav1.Duration      `json:"duration,omitempty"`
+	MaxRestarts  *intstr.IntOrString   `json:"maxRestarts,omitempty"`
```

Comment:
I'm wondering if it is necessary to have a `*intstr.IntOrString `. I feel like `*int` should be enough.